baybe/utils/chemistry.py

```diff
@@ -72,158 +72,117 @@ def name_to_smiles(name: str) -> str:
 
 @lru_cache(maxsize=None)
```

Comment:
how do I check that?

Reply:
Let's go step by step. Right now, there are a few things that seem really fishy to me, perhaps you can comment on that first:
* The `Chem.PropertyMol.PropertyMol` does not exist. What is that supposed to be?
* I suspect that second input type should capture the case when a `mol_list` is passed instead? But according to the types defined in skfp, that should simply be a `list[Mol]`, no?
* You call `fingerprint_encoder.transform([molecule])`, but according to the defined types, that transform should fail when a `mol_list` is passed, because it would then be wrapped into another list
* Which brings me to the most important question: have you actually tested the conformer case somewhere? If not, I think we could simply use a conformer example in the userguide. This would have both the advantage that the interface of `kwargs_conformer` gets demonstrated + the code is automatically tested there

Reply:
The input is always one molecule in the function I made, never a list - as per the type hints. We could change it to list if desired. If I remember correctly, I made it that way because before it was also a function with single molecule input.

It inputs either str SMILES or the conformer object (required by some fps). The conformer is of the [PropertyMol] (https://www.rdkit.org/docs/source/rdkit.Chem.PropertyMol.html) class, but I am unsure why it is shown as not existent.
```
smiles=['CCO']
mol_list = ConformerGenerator().transform(
    MolFromSmilesTransformer().transform(smiles)
)
mol_list
[<rdkit.Chem.PropertyMol.PropertyMol at 0x348498cc0>]
```

Test is here for all fingerprints: tests/test_fingerprints.py

Reply:
Ah OK, sorry, I had a wrong look at the part with the list. Now I see that `PropertyMol` is just a subclass of `Mol` and that the double list layer actually does not occur. Hence, second and third bullet are obsolete. Fourth is also obsolete now that I realize that obviously the test also includes conformers 👍🏼 

But bullet one remains. Indeed, the class exists and the import path is correct. So honestly, I have no clue what is the problem here. My first suspicion was that is has something to do with our optional import, but you in fact get the same error when importing from `rdkit` directly. Hm .... 🤔 

The simply solution would be to add a type ignore statement here but that feels really unsatisfying. Would you mind having a quick look into it? Perhaps we can find the root cause

Reply:
the problem of `seemingly non existing imports` is common in rdkit, i think its because this package is strongly intertwinded with C code, an ignore should suffice

Reply:
I think I have a better solution: why not simply replace the annotation with `Chem.Mol`? The latter can be found + it is anyway the more general input type here. Give me thumbs up and I will change

Reply:
Change implement. Resolve if agree

Reply:
i think it should be fine, type hint of `transform` seems to also only require a `Mol` so im not sure why it was set to `PropertyMol` in the first place? @Hrovatin

Reply:
I set it based on what was output of the conformer class, but yes, we can make it more general.